From 4027f11b113315c75cb0387fc247b3dfdbe9a237 Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Mon, 4 Mar 2019 17:22:51 +0800
Subject: [PATCH 07/15] qwaylandwindow: Don't try to move fullscreen/maximized
 window

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 src/client/qwaylandwindow.cpp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/client/qwaylandwindow.cpp b/src/client/qwaylandwindow.cpp
index a8d6ad52..b9b0bd9e 100644
--- a/src/client/qwaylandwindow.cpp
+++ b/src/client/qwaylandwindow.cpp
@@ -352,7 +352,11 @@ void QWaylandWindow::setGeometry_helper(const QRect &rect)
 
 void QWaylandWindow::setGeometry(const QRect &rect)
 {
-    setGeometry_helper(rect);
+    if (window()->windowStates() == Qt::WindowFullScreen
+        || window()->windowStates() == Qt::WindowMaximized)
+        setGeometry_helper(QRect(QPoint(), rect.size()));
+    else
+        setGeometry_helper(rect);
 
     if (window()->isVisible() && rect.isValid()) {
         if (mWindowDecoration)
-- 
2.20.1

