From 4dbb960d84f1fa24a33d0922611979e570c1db72 Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Tue, 13 Oct 2020 17:01:39 +0800
Subject: [PATCH 13/15] qwaylandwindow: Clear mWaitingToApplyConfigure when
 reset

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 src/client/qwaylandwindow.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/client/qwaylandwindow.cpp b/src/client/qwaylandwindow.cpp
index b432589d..c129de9f 100644
--- a/src/client/qwaylandwindow.cpp
+++ b/src/client/qwaylandwindow.cpp
@@ -249,6 +249,9 @@ void QWaylandWindow::reset(bool sendDestroyEvent)
         QPlatformSurfaceEvent e(QPlatformSurfaceEvent::SurfaceAboutToBeDestroyed);
         QGuiApplication::sendEvent(window(), &e);
     }
+
+    mWaitingToApplyConfigure = false;
+
     delete mShellSurface;
     mShellSurface = nullptr;
     delete mSubSurfaceWindow;
-- 
2.20.1

