From 84f0b819e1cf21022d34e64f886e8a57e91d94c0 Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Fri, 30 Nov 2018 17:50:55 +0800
Subject: [PATCH 10/18] qwindow: Add interface to override win id for video
 window

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 src/gui/kernel/qwindow.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/gui/kernel/qwindow.h b/src/gui/kernel/qwindow.h
index 5ee1d00f5..b2d652b9f 100644
--- a/src/gui/kernel/qwindow.h
+++ b/src/gui/kernel/qwindow.h
@@ -402,6 +402,13 @@ template <> inline const QWindow *qobject_cast<const QWindow*>(const QObject *o)
 Q_GUI_EXPORT QDebug operator<<(QDebug, const QWindow *);
 #endif
 
+// Inherit this to overrid window id for video window
+class Q_GUI_EXPORT QVideoWindowAbstractInterface
+{
+public:
+    virtual WId videoWinId() const = 0;
+};
+
 QT_END_NAMESPACE
 
 #endif // QWINDOW_H
-- 
2.20.1

