From 0908bd5bf2909beaee6174a082396535015ee739 Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Wed, 1 Apr 2020 14:13:03 +0800
Subject: [PATCH 15/18] qsurfaceformat: Set default major version to 3

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 src/gui/kernel/qsurfaceformat.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/gui/kernel/qsurfaceformat.cpp b/src/gui/kernel/qsurfaceformat.cpp
index 238886220..b92b26961 100644
--- a/src/gui/kernel/qsurfaceformat.cpp
+++ b/src/gui/kernel/qsurfaceformat.cpp
@@ -70,7 +70,7 @@ public:
         , numSamples(-1)
         , renderableType(QSurfaceFormat::DefaultRenderableType)
         , profile(QSurfaceFormat::NoProfile)
-        , major(2)
+        , major(3)
         , minor(0)
         , swapInterval(1) // default to vsync
         , colorSpace(QSurfaceFormat::DefaultColorSpace)
-- 
2.20.1

