From 7b1aaf11b26c3908d5fb4cb43822734dfec07260 Mon Sep 17 00:00:00 2001
From: Andrey Konovalov <andrey.konovalov@linaro.org>
Date: Tue, 9 Apr 2019 21:32:36 +0300
Subject: [PATCH 18/60] waylandsink: make gst_wl_window_is_toplevel aware of
 XDG shell

---
 ext/wayland/wlwindow.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/ext/wayland/wlwindow.c b/ext/wayland/wlwindow.c
index 0a1c15601..77b2d81d4 100644
--- a/ext/wayland/wlwindow.c
+++ b/ext/wayland/wlwindow.c
@@ -360,7 +360,10 @@ gst_wl_window_is_toplevel (GstWlWindow * window)
 {
   g_return_val_if_fail (window != NULL, FALSE);
 
-  return (window->wl_shell_surface != NULL);
+  if (window->display->xdg_wm_base)
+    return (window->xdg_toplevel != NULL);
+  else
+    return (window->wl_shell_surface != NULL);
 }
 
 static void
-- 
2.20.1

