From 41ee754df1d773c632ae5e590ff0a7878056e08d Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Fri, 30 Nov 2018 17:45:33 +0800
Subject: [PATCH 3/3] qquickwidget: Provide widget window id for video window

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 src/quickwidgets/qquickwidget.cpp | 16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

diff --git a/src/quickwidgets/qquickwidget.cpp b/src/quickwidgets/qquickwidget.cpp
index 39780f8d..ddf56f3b 100644
--- a/src/quickwidgets/qquickwidget.cpp
+++ b/src/quickwidgets/qquickwidget.cpp
@@ -100,12 +100,26 @@ private:
     QQuickWidget *m_quickWidget;
 };
 
+class QQuickOffscreenWindow : public QQuickWindow, public QVideoWindowAbstractInterface
+{
+public:
+    QQuickOffscreenWindow(QQuickWindowPrivate &dd, QQuickRenderControl *control)
+        : QQuickWindow(dd, control),
+        renderControl(control) {}
+    WId videoWinId() const Q_DECL_OVERRIDE {
+        QWindow *window = renderControl->renderWindow(NULL);
+        return window ? window->winId() : 0;
+    }
+private:
+    QQuickRenderControl *renderControl;
+};
+
 void QQuickWidgetPrivate::init(QQmlEngine* e)
 {
     Q_Q(QQuickWidget);
 
     renderControl = new QQuickWidgetRenderControl(q);
-    offscreenWindow = new QQuickWindow(*new QQuickOffcreenWindowPrivate(),renderControl);
+    offscreenWindow = new QQuickOffscreenWindow(*new QQuickOffcreenWindowPrivate(),renderControl);
     offscreenWindow->setTitle(QString::fromLatin1("Offscreen"));
     offscreenWindow->setObjectName(QString::fromLatin1("QQuickOffScreenWindow"));
     // Do not call create() on offscreenWindow.
-- 
2.20.1

