From 986c77ae73a411935fa5e62aeb584e0a606c56af Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Fri, 30 Nov 2018 17:54:38 +0800
Subject: [PATCH 05/14] gsttools: videooverlay: Support overriding video win id

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 src/gsttools/qgstreamervideooverlay.cpp | 19 ++++++++++++-------
 1 file changed, 12 insertions(+), 7 deletions(-)

diff --git a/src/gsttools/qgstreamervideooverlay.cpp b/src/gsttools/qgstreamervideooverlay.cpp
index 769c53e..3621969 100644
--- a/src/gsttools/qgstreamervideooverlay.cpp
+++ b/src/gsttools/qgstreamervideooverlay.cpp
@@ -458,6 +458,15 @@ static QWindow *findWindow(WId id) {
     return NULL;
 }
 
+static QWindow *getVideoWindow(WId id) {
+    QWindow *window = findWindow(id);
+
+    QVideoWindowAbstractInterface *intf =
+        dynamic_cast<QVideoWindowAbstractInterface *>(window);
+
+    return intf ? findWindow(intf->videoWinId()) : window;
+}
+
 void QGstreamerVideoOverlay::setWindowHandle_helper(WId id)
 {
 #ifdef ENABLE_WAYLAND_PLATFORM
@@ -480,7 +489,7 @@ void QGstreamerVideoOverlay::setWindowHandle_helper(WId id)
 
     overlay = GST_VIDEO_OVERLAY(m_videoSink);
 
-    window = findWindow(id);
+    window = getVideoWindow(id);
     if (!window) {
         gst_video_overlay_set_window_handle(overlay, 0);
         goto set_rectangle;
@@ -494,12 +503,8 @@ void QGstreamerVideoOverlay::setWindowHandle_helper(WId id)
 
     // It's wayland platform, using wl_surface as window handle.
     if (compositor) {
-        if (!surface || !isActive()) {
-            gst_video_overlay_set_window_handle(overlay, 0);
-            goto out;
-        }
-
-        gst_video_overlay_set_window_handle(overlay, (WId) surface);
+        gst_video_overlay_set_window_handle(overlay,
+                                            isActive() ? (WId) surface : 0);
 
         if (m_rect.width() <= 0 || m_rect.height() <= 0)
             goto out;
-- 
2.20.1

