From 0d9d3e788c1f5e19859394644d7c6d38fa71122e Mon Sep 17 00:00:00 2001
From: Frank Liu <frank.liu@rock-chips.com>
Date: Tue, 31 Mar 2020 16:32:49 +0800
Subject: [PATCH 19/19] Fix rkmpp decoder failed for rv1108 platform

Signed-off-by: Frank Liu <frank.liu@rock-chips.com>
---
 libavcodec/rkmppdec.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/libavcodec/rkmppdec.c b/libavcodec/rkmppdec.c
index f4e03e2..08ea774 100644
--- a/libavcodec/rkmppdec.c
+++ b/libavcodec/rkmppdec.c
@@ -264,6 +264,10 @@ static int rkmpp_init_decoder(AVCodecContext *avctx)
     RK_S64 paramS64;
     RK_S32 paramS32;
 
+#ifdef CONFIG_ION
+    avctx->pix_fmt = AV_PIX_FMT_NONE;
+#endif
+
     if (avctx->pix_fmt == AV_PIX_FMT_NONE &&
         avctx->sw_pix_fmt == AV_PIX_FMT_NONE) {
         // chromium only support AV_PIX_FMT_YUV420P
-- 
2.17.1

