From 8334e03407d42caca79f6c2f6dfa71e0ca9dcac1 Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Wed, 26 Dec 2018 17:51:51 +0800
Subject: [PATCH 08/13] gstreamer: mediaplayer: Fix hangs when replacing paused
 video sink

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 .../gstreamer/mediaplayer/qgstreamerplayersession.cpp        | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/plugins/gstreamer/mediaplayer/qgstreamerplayersession.cpp b/src/plugins/gstreamer/mediaplayer/qgstreamerplayersession.cpp
index aa8ce5a..8dceb25 100755
--- a/src/plugins/gstreamer/mediaplayer/qgstreamerplayersession.cpp
+++ b/src/plugins/gstreamer/mediaplayer/qgstreamerplayersession.cpp
@@ -675,6 +675,11 @@ void QGstreamerPlayerSession::setVideoRenderer(QObject *videoOutput)
         //while the sink is paused. The pad will be blocked as soon as the current
         //buffer is processed.
         if (m_state == QMediaPlayer::PausedState) {
+            if (!m_everPlayed) {
+                finishVideoOutputChange();
+                seek(m_lastPosition);
+            }
+
 #ifdef DEBUG_PLAYBIN
             qDebug() << "Starting video output to avoid blocking in paused state...";
 #endif
-- 
2.20.1

