From 2e161af98e24ad4a7e0e235fc59ea13042a237dc Mon Sep 17 00:00:00 2001
From: "jkand.huang" <jkand.huang@rock-chips.com>
Date: Tue, 2 Jun 2020 17:41:14 +0800
Subject: [PATCH] fix rtcp rtp-timestamp error.

Signed-off-by: jkand.huang <jkand.huang@rock-chips.com>
---
 liveMedia/RTPSink.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/liveMedia/RTPSink.cpp b/liveMedia/RTPSink.cpp
index cb2572a..a3c9b75 100644
--- a/liveMedia/RTPSink.cpp
+++ b/liveMedia/RTPSink.cpp
@@ -108,6 +108,11 @@ u_int32_t RTPSink::presetNextTimestamp() {
     fNextTimestampHasBeenPreset = True;
   }
 
+  // fix rtcp rtp-timestamp error.
+  struct timeval timeNow_reset;
+  gettimeofday(&timeNow_reset, NULL);
+  convertToRTPTimestamp(timeNow_reset);
+
   return tsNow;
 }
 
-- 
2.26.1

